src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/ChainPal.Apple.cs

```diff
@@ -38,9 +39,16 @@ internal SecTrustChainPal()
             X509Certificate2Collection? customTrustStore,
             X509ChainTrustMode trustMode)
         {
+            _certificateAuthorityStore?.Dispose();
+            _certificateAuthorityStore = OpenCertificateAuthorityStore();
```

Comment:
My biggest question here is the lifetime of the X509Store and its certificates are managed correctly here. I think this is correct, but, could use some input here. My concerns largely are making sure the cert handles are cleaned up correctly, and likely not freed before we are done with them.

And of course ideas for tests that prove that out are certainly welcome.